src/main/java/hudson/plugins/throttleconcurrents/ThrottleJobProperty.java

```diff
@@ -229,29 +243,82 @@ public boolean isThrottleMatrixConfigurations() {
         }
         for (ThrottleJobProperty t : properties) {
             if (t.getThrottleEnabled()) {
                 if (t.getCategories() != null && t.getCategories().contains(category)) {
-                    Job<?,?> p = t.owner;
+                    Job<?, ?> p = t.owner;
                     if (/*is a task*/ p instanceof Queue.Task && /* not deleted */getItem(p.getParent(), p.getName()) == p &&
                         /* has not since been reconfigured */ p.getProperty(ThrottleJobProperty.class) == t) {
                         categoryTasks.add((Queue.Task) p);
                         if (p instanceof MatrixProject && t.isThrottleMatrixConfigurations()) {
-                            for (MatrixConfiguration mc : ((MatrixProject)p).getActiveConfigurations()) {
+                            for (MatrixConfiguration mc : ((MatrixProject) p).getActiveConfigurations()) {
                                 categoryTasks.add(mc);
                             }
                         }
                     }
                 }
             }
         }
+
         return categoryTasks;
     }
+
+    static List<ThrottledStepInfo> getThrottledPipelinesForCategory(String category) {
+        List<ThrottledStepInfo> throttledPipelines = new ArrayList<>();
+
+        DescriptorImpl descriptor = Jenkins.getActiveInstance().getDescriptorByType(DescriptorImpl.class);
+        for (Map.Entry<String,Integer> currentPipeline : descriptor.getThrottledPipelinesForCategory(category).entrySet()) {
+            Run<?,?> flowNodeRun = Run.fromExternalizableId(currentPipeline.getKey());
+
+            if (flowNodeRun == null) {
+                // No run found, so remove the throttle.
+                descriptor.removeThrottledPipelineForCategory(currentPipeline.getKey(), category, null);
+            } else if (!(flowNodeRun instanceof FlowExecutionOwner.Executable)) {
+                // If for some reason we've somehow ended up with a non-pipeline job, remove the throttle.
+                descriptor.removeThrottledPipelineForCategory(currentPipeline.getKey(), category, null);
+            } else if (!flowNodeRun.isBuilding()) {
+                // The run is done building, so remove the throttle.
+                descriptor.removeThrottledPipelineForCategory(currentPipeline.getKey(), category, null);
+            } else {
+                FlowExecutionOwner owner = ((FlowExecutionOwner.Executable)flowNodeRun).asFlowExecutionOwner();
+                FlowExecution execution = owner.getOrNull();
+                if (execution == null) {
+                    // For some reason, the flow execution is null, so again? Remove the throttle.
+                    descriptor.removeThrottledPipelineForCategory(currentPipeline.getKey(), category, null);
+                } else {
+                    try {
+                        for (StepExecution stepExec : execution.getCurrentExecutions(false).get()) {
```

Comment:
I am not sure why you think you need to do this but it has a very bad smell.

If you really have no `FlowNode` or `StepContext` that you are starting with, probably you want to pass `true`, since the innermost step should have the `Computer` in context if it somewhere inside `node`.

But the fact that you do _not_ have a specific step in mind is alarming. What is this code doing when you have a dozen `node` blocks running in `parallel`?

Did you rather mean to pay attention to the actual `ExecutorStepExecution.PlaceholderTask` being scheduled and look up its `context`? Currently there is no supported API for extracting contextual objects out of such a `Task`, so we would need to define one (cf. discussion in https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/26).